src/main/java/org/openlmis/notification/service/UserContactDetailsService.java

```diff
@@ -48,6 +71,57 @@ public UserContactDetails addOrUpdate(UserContactDetails details) {
         : addUserContactDetails(details);
   }
 
+  /**
+   * Saves user contact details.
+   *
+   * @param userDto user contact details object
+   * @param successfulResults list with success results
+   * @param failedResults list with failed results
+   */
+  public void saveUsersContactDetails(UserContactDetailsDto userDto,
+      List<UserContactDetailsResponseDto.UserDetailsResponse> successfulResults,
+      List<UserContactDetailsResponseDto.FailedUserDetailsResponse> failedResults
+  ) {
+    try {
+      BindingResult bindingResult = new BeanPropertyBindingResult(userDto, "userContactDetailsDto");
+      permissionService.canManageUserContactDetails(userDto.getReferenceDataUserId());
```

Comment:
What I meant is that the check for a role could be done only once before the whole batch, **not** for every item.